ui/app/js/directives/permission_options_directive.js

```diff
@@ -0,0 +1,127 @@
+'    <p class="help-block" ng-show="errors.options">{{ errors.options.join(\', \') }}</p>'+
+'</div>'+
+'<div class="form-group" ng-show="!!(selected_options.indexOf(\'actions\')+1)">'+
+'    <label><i>actions</i></label>'+
+'    <div class="row-fluid m-tag__input">'+
+'        <label ng-repeat="action in selected_actions" class="btn btn-sm btn-warning">{{ action }} <span class="badge" ng-click="actionRemove(action)">x</span></label>'+
+'        <input type="text" typeahead="action for action in actions[permission.permission] | filter:$viewValue | limitTo:16" ng-model="action"></input>'+
+'    </div>'+
+'</div>'+
+'<div class="form-group" ng-show="!!(selected_options.indexOf(\'products\')+1)">'+
+'    <label><i>products</i></label>'+
+'    <div class="row-fluid m-tag__input">'+
+'        <label ng-repeat="product in selected_products" class="btn btn-sm btn-primary">{{ product }} <span class="badge" ng-click="productRemove(product)">x</span></label>'+
+'        <input type="text" typeahead="product for product in products | filter:$viewValue | limitTo:16" ng-model="product"></input>'+
+'    </div>'+
+'</div> ',
+        controller: function($scope, Releases){
+            var allPermissions = {
+                "admin": ["products"],
+                "release": ["actions", "products"],
+                "release_locale": ["actions", "products"],
+                "release_read_only": ["actions", "products"],
+                "rule": ["actions", "products"],
+                "permission": ["actions"],
+                "required_signoff": ["products"],
+                "scheduled_change": ["actions"],
+            };
+            $scope.$watch('permission.permission',function(){
+                $scope.available_options = allPermissions[$scope.permission.permission];
+
+                // clear  all previous selections 
+                $scope.selected_options = [];
+                $scope.selected_actions = [];
+                $scope.selected_products = [];
+            });
+            $scope.selected_options = [];
+            $scope.actions = {
+                "admin": [],
+                "rule": ["create", "modify", "delete"],
+                "release": ["create", "modify", "delete"],
+                "release_read_only": ["set", "unset"],
+                "release_locale": ["modify"],
+                "required_signoff": ["create", "modify", "delete"],
+                "permission": ["create", "modify", "delete"],
+                "scheduled_change": ["enact"],
+                "":[]
+            };
+
+            $scope.selected_actions = [];
+            $scope.action = null;
+
+            $scope.products = [];
+            $scope.selected_products = [];
+            $scope.product = null;
+
+            // get products in releases
+            Releases.getProducts().success(function(response) {
+                $scope.products = response.product;
+            });
+
+            $scope.$watch('[ action, product ] | json',function(){
```

Comment:
What does the `| json` do here?

Reply:
Tried `$scope.$watch('[ action, product ]', function() {}` and wasn't working. Got the idea from SO  https://stackoverflow.com/questions/11952579/watch-multiple-scope-attributes#answer-14410060

Reply:
OK, so it's clear that it's performing some useful function - but it would be good to understand exactly what.

Reply:
@mozbhearsum Is it ok if I just add a comment for that ?

Reply:
Yes...but the current comment doesn't really answer the question. Is the `[...] | json` structure needed because you're watching multiple variables instead of just one? That's the part I don't understand.

Reply:
@mozbhearsum  Yeah exactly, It is because I'm watching both `action` and `product` at the same time.